vms/platformvm/cache_locked_utxos_chain_state_test.go

```diff
@@ -0,0 +1,367 @@
+			msg:           "Should have failed because UTXOs consumed are already bonded",
+		},
+		{
+			name: "Consumed UTXOs already deposited test",
+			args: args{
+				inputs: []*avax.TransferableInput{
+					{
+						Asset: avax.Asset{ID: vm.ctx.AVAXAssetID},
+						In: &secp256k1fx.TransferInput{
+							Amt: 1,
+						},
+					},
+				},
+				bond: false,
+			},
+			utxoLockState: utxoLockState{BondTxID: nil, DepositTxID: &ids.ID{}},
+			wantErr:       true,
+			msg:           "Should have failed because UTXOs consumed are already deposited",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cs := &lockedUTXOsChainStateImpl{
+				bonds:        make(map[ids.ID]ids.Set),
+				deposits:     make(map[ids.ID]ids.Set),
+				lockedUTXOs:  make(map[ids.ID]utxoLockState),
+				updatedUTXOs: make(map[ids.ID]utxoLockState),
+			}
+			cs.lockedUTXOs[tt.args.inputs[0].InputID()] = tt.utxoLockState
+			err := cs.SemanticVerifyLockInputs(tt.args.inputs, tt.args.bond)
+			assert.Equal(t, tt.wantErr, err != nil, tt.msg)
+		})
+	}
+}
+
+func TestUpdateUTXOs(t *testing.T) {
+	type fields struct {
+		bonds        map[ids.ID]ids.Set
+		deposits     map[ids.ID]ids.Set
+		lockedUTXOs  map[ids.ID]utxoLockState
+		updatedUTXOs map[ids.ID]utxoLockState
+	}
+	type args struct {
+		updatedUTXOStates map[ids.ID]utxoLockState
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    lockedUTXOsChainState
+		wantErr bool
+		msg     string
+	}{
+		{
+			name: "Happy path bonding",
+			fields: fields{
+				bonds:        make(map[ids.ID]ids.Set),
+				deposits:     make(map[ids.ID]ids.Set),
+				lockedUTXOs:  make(map[ids.ID]utxoLockState),
+				updatedUTXOs: make(map[ids.ID]utxoLockState),
+			},
+			args: args{updatedUTXOStates: map[ids.ID]utxoLockState{{1, 1}: {
+				BondTxID:    &ids.ID{9, 9},
+				DepositTxID: nil,
+			}}},
+			want: &lockedUTXOsChainStateImpl{
+				bonds: map[ids.ID]ids.Set{
+					{9, 9}: map[ids.ID]struct{}{{1, 1}: {}},
+				},
+				deposits: map[ids.ID]ids.Set{},
+				lockedUTXOs: map[ids.ID]utxoLockState{{1, 1}: {
+					BondTxID:    &ids.ID{9, 9},
+					DepositTxID: nil,
+				}},
+				updatedUTXOs: map[ids.ID]utxoLockState{{1, 1}: {
+					BondTxID:    &ids.ID{9, 9},
+					DepositTxID: nil,
+				}},
+			},
+			wantErr: false,
+			msg:     "Happy path bonding",
+		},
+		{
+			name: "Happy path unbonding",
+			fields: fields{
+				bonds: map[ids.ID]ids.Set{
+					{9, 9}: map[ids.ID]struct{}{{1, 1}: {}},
+				},
+				deposits: make(map[ids.ID]ids.Set),
+				lockedUTXOs: map[ids.ID]utxoLockState{{1, 1}: {
+					BondTxID:    &ids.ID{9, 9},
+					DepositTxID: nil,
+				}},
+				updatedUTXOs: map[ids.ID]utxoLockState{{1, 1}: {
+					BondTxID:    &ids.ID{9, 9},
+					DepositTxID: nil,
+				}},
+			},
+			args: args{updatedUTXOStates: map[ids.ID]utxoLockState{{1, 1}: {
+				BondTxID:    nil,
+				DepositTxID: nil,
+			}}},
+			want: &lockedUTXOsChainStateImpl{
+				bonds: map[ids.ID]ids.Set{
```

Comment:
bonds should be map[ids.ID]ids.Set{}

Reply:
same for deposit test case

Reply:
When unbonding or undepositing, if i understood correctly, only the value of the bonds and deposit maps is being removed. Should the Tx.ID be also removed? If yes, i think it is not working like this.

Reply:
Looks like you'r right, gonna push fix for that